Sources/HaCWebsiteLib/Events/HackathonEvent.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+struct HackathonEvent : Event {
```

Comment:
I like that we're creating different event types here. I think it would be valuable for us to figure out what these should be in advance.

Reply:
So we may need:
* HaC Nights
* Reading Groups
* Open Meetings 
* Q&As 
* ??
I don't necessarily think that all of these need their own types but maybe we should create a default class?

Reply:
What is the justification for these having separate types?

Reply:
Well we have different types for Workshops and Hackathons at the moment because each carries different information. I'm not sure there's a huge amount of justification for creating many more types beyond this though.

Reply:
Well, we have different types for Workshops and Hackathons at the moment because each carries different information. I'm not sure there's a huge amount of justification for creating many more types beyond this though.

Reply:
Maybe we couple the names slightly more to the data they hold so we don't need so many representations?

- `GeneralEvent` (for an event with a stock-standard page. Could fit open meetings, reading groups, Q&As, talks, etc...)
- `WorkshopEvent` (event linked to a workshop. Maybe a special case of `GeneralEvent`)
- `ExternalWebsiteEvent` (event linked to an external website such as Hack Cambridge)

If we weren't storing these all in a database, I'd be in favour of going crazy adding new types. Unfortunately these are likely all going to be in one DB table, so we may have to be a little pragmatic.